src/backend/joanie/signature/backends/lex_persona/client.py

```diff
@@ -0,0 +1,400 @@
+
+from joanie.core.utils import webhooks
+
+from ..base import BaseSignatureBackend
+
+logger = getLogger(__name__)
+
+
+# pylint: disable=too-many-instance-attributes
+class LexPersonaClient(BaseSignatureBackend):
+    """
+    The Signature base class.
+    It contains generic methods to trigger
+    the workflow process of signing a document,
+    and to retrieve it once it has been signed.
+    """
+
+    name = "lex_persona"
+
+    def __init__(self, configuration):
+        """
+        Access data must be kept secret.
+        Replace those values in your .env file with your
+        own credentials provided by Lex Persona.
+        """
+        super().__init__(configuration)
+        self._set_configuration_attributes(configuration)
+        created, validity = self._set_create_and_validity_date()
+        self.workflow_id = str()
+        self.create_payload = {
+            "name": "Contract Definition / Convention de formation",
+            "description": "A simple document that engages the student to follow the course",
+            "steps": [
+                {
+                    "stepType": "signature",
+                    "recipients": [],
+                    "requiredRecipients": 1,
+                    "validityPeriod": validity,
+                    "invitePeriod": created,
+                    "maxInvites": 0,
+                    "sendDownloadLink": True,
+                    "allowComments": True,
+                    "hideAttachments": False,
+                    "hideWorkflowRecipients": True,
+                }
+            ],
+            "notifiedEvents": [
+                "recipientRefused",
+                "workflowFinished",
+            ],
+            "watchers": [],
+        }
+
+    def handle_notification(self, request):
+        """
+        Method triggered when a notification is sent by the signature provider.
+        If the response.method is POST and the webhook event id is
+        verified, then we can proceed to the webhook synchronize
+        signature method.
+        """
+        response = None
+
+        if request.method == "POST":
+            data = json.loads(request.body)
+            try:
+                webhook_event_id = data["id"]
+                response = self._verify_webhook_event(webhook_event_id)
+            except KeyError as error:
+                logger.error("There is no ID key in the request body :%s", error)
+                return Response(
+                    {"error": "Failed, missing 'id' information in request body."},
+                    status=404,
+                )
+
+        if response is not None and response.status_code == 404:
+            return response
+        if response is not None and response.status_code == 200:
+            return webhooks.synchronize_signed_document(request)
+
+        return Response(
+            {"error": "Unknown error occurred"},
+            status=status.HTTP_500_INTERNAL_SERVER_ERROR,
```

Comment:
Returning the same error that the one present in response.status_code might be better
500 with the unknown error message should be reserved for the case where response is None

Reply:
Indeed, I made some adjustment : 
```python
if request.method == "POST":
    data = json.loads(request.body)
    try:
        webhook_event_id = data["id"]
        response = self._verify_webhook_event(webhook_event_id)
    except KeyError as error:
        logger.error("There is no ID key in the request body :%s", error)
        return Response(
            {"error": "Failed, missing 'id' information in request body."},
            status=status.HTTP_404_NOT_FOUND,
        )
    if response.status_code == status.HTTP_404_NOT_FOUND:
        return response
    if response.status_code == status.HTTP_200_OK:
        return webhooks.synchronize_signed_document(request)
return Response(
    {"error": "Invalid request method"},
    status.HTTP_405_METHOD_NOT_ALLOWED,
)
```
What do you think ? 
I made sure to indent the code where we reach response.status_code with a valid return from the API.
If the method that was used is not meeting my condition, then it raises a 405 error.